lib/everest.cpp

```diff
@@ -158,30 +158,33 @@ Everest::Everest(std::string module_id_, const Config& config_, bool validate_da
     }
 
     // setup error_databases, error_managers and error_state_monitors for all requirements
-    for (const Requirement& req : config.get_requirements(module_id)) {
-        // setup shared database
-        std::shared_ptr<error::ErrorDatabaseMap> error_database = std::make_shared<error::ErrorDatabaseMap>();
-
-        // setup error manager
-        std::string interface_name = this->module_manifest.at("requires").at(req.id).at("interface");
-        json interface_def = this->config.get_interface_definition(interface_name);
-        std::list<std::string> allowed_error_types;
-        for (const auto& error_namespace_it : interface_def["errors"].items()) {
-            for (const auto& error_name_it : error_namespace_it.value().items()) {
-                allowed_error_types.push_back(error_namespace_it.key() + "/" + error_name_it.key());
+    for (const auto& fulfillments : config.get_fulfillments(module_id)) {
+        for (const Fulfillment& fulfillment : fulfillments.second) {
+            auto& req = fulfillment.requirement;
+            // setup shared database
+            std::shared_ptr<error::ErrorDatabaseMap> error_database = std::make_shared<error::ErrorDatabaseMap>();
+
+            // setup error manager
+            std::string interface_name = this->module_manifest.at("requires").at(req.id).at("interface");
+            json interface_def = this->config.get_interface_definition(interface_name);
+            std::list<std::string> allowed_error_types;
+            for (const auto& error_namespace_it : interface_def["errors"].items()) {
```

Comment:
these changes are probably not necessary, if you keep `get_requirements`

Reply:
I've re-added get_requirements, that does lead to some code duplication though since information required for the Fulfillments is not available from this function